pkg/apiserver/request/requestinfo.go

```diff
@@ -204,7 +204,22 @@ func (r *RequestInfoResolver) GetRequestInfo(req *http.Request) (RequestInfo, er
 		requestInfo.Verb = "deletecollection"
 	}
 
-	return requestInfo, nil
+	return &requestInfo, nil
+}
+
+type requestInfoKeyType int
+
+const requestInfoKey requestInfoKeyType = iota
```

Comment:
That's why I was asking on irc yesterday about moving `Context` out of `pkg/api`. But keys are interfaces and interfaces are equal if the type and the value are equal. So this private type idea actually works.

Will add a comment.